compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3588,4 +3588,19 @@ public boolean isOrderedType(BType type) {
                 return isSimpleBasicType(type.tag);
         }
     }
+
+    public boolean isUnionOfSimpleBasicTypes(BType type) {
+        switch (type.tag) {
```

Comment:
Do we need a switch case here? We can stick to a simple, single if check and a default return

Reply:
Changed.